boards/nxp/ls1046ardb/board.yml

```diff
@@ -0,0 +1,8 @@
+board:
+  name: nxp_ls1046ardb
+  vendor: nxp
+  socs:
+  - name: ls1046a
+    variants:
+    - name: smp_2cores
+    - name: smp_4cores
```

Comment:
Going by what we have in other boards and https://github.com/zephyrproject-rtos/zephyr/pull/69400/commits/b3720c7bf78bb5dc72e6a86a6b0570040988bd8a can you change this so smp is a variant which enables either 2 or 4 cores then have the other as a variant of that, so the boards would be `nxp_ls1046ardb/ls1046a`, `nxp_ls1046ardb/ls1046a/smp`, and (`nxp_ls1046ardb/ls1046a/smp/2cores` or `nxp_ls1046ardb/ls1046a/smp/4cores`)

Reply:
Changed, please review again, thanks.

Reply:
@nordicjm, I updated but I found currently there is document building failure, but I don't know why we have the failure, it seems it is not caused by this PR, thanks.

Reply:
It is caused by this PR, you need to remove the old arm64 boards_legacy documentation index file if the folder is empty and remove that file being listed by the index.rst file

Reply:
Rebased and added a new commit to clean up empty directory: https://github.com/zephyrproject-rtos/zephyr/pull/69416/commits/cb4fecf9bb826bf01db32f0568dc30c46b4e0d04